evm_loader/program/src/executor/state.rs

```diff
@@ -285,17 +285,36 @@ impl<'a, B: AccountStorage> Database for ExecutorState<'a, B> {
     }
 
     async fn code_hash(&self, from_address: Address, chain_id: u64) -> Result<[u8; 32]> {
-        use solana_program::keccak::hash;
-
-        for action in &self.actions {
-            if let Action::EvmSetCode { address, code, .. } = action {
-                if &from_address == address {
-                    return Ok(hash(code).to_bytes());
-                }
-            }
+        // https://eips.ethereum.org/EIPS/eip-1052
+        // https://eips.ethereum.org/EIPS/eip-161
+
+        macro_rules! data_account_exists {
+            ($self:ident, $address:ident, $chain_id:ident) => {
+                $self.nonce($address, $chain_id).await? > 0
+                    || $self.balance($address, $chain_id).await? > 0
+            };
         }
```

Comment:
Makes if statement easier to read. Could use closure but wanted to pass the arguments explicitly and got lifetime issues. A macro looked cleanest.

Reply:
I prefer not to have a function or macro for this at all. It only complicates things.

Reply:
If you feel strongly about this, I will change it to your preference.

Reply:
I also think just inlining the method and adding a comment is enough.

Reply:
As someone coming in fresh, I would have greatly appreciated if the code was written in the way that I propose. However, I will take your advice on how to implement it here.